Pass stack trace to generated split graph module during model splitting

zijianshen opened the pull request:
Summary: For debugging purpose, we may want to maintain the stack trace in the nodes of graph module.

Test Plan: More test plans in D68132673

Reviewed By: faran928

Differential Revision: D68302850




cc @ezyang @SherlockNoMad @EikanWang @jgong5 @wenzhe-nrv

ZhengkaiZ commented on torch/fx/passes/split_utils.py:
```diff
@@ -293,6 +295,10 @@ def remap_func(x):
     main_g._codegen = gm.graph._codegen
+    
+    for n, m in zip(main_g.nodes, gm.graph.nodes):
+        if "stack_trace" in m.meta:
```
could we create small utils function to copy node stack_trace?

zijianshen committed:
Pass stack trace to generated split graph module during model splitting (#145033)

Summary:

For debugging purpose, we may want to maintain the stack trace in the nodes of graph module.

Test Plan: More test plans in D68132673

Reviewed By: faran928

Differential Revision: D68302850
